libcaf_net/caf/net/http/client.cpp

```diff
@@ -206,10 +206,49 @@ class client_impl : public client {
       input.subspan(payload_len_);
       mode_ = mode::read_header;
       return consumed;
+    } else if (mode_ == mode::read_chunks) {
```

Comment:
That -1 way down below should basically be unreachable. I'd prefer an explicit "sanity check" here.

```c++
    }
    if (mode_ != mode::read_chunks) {
      abort_and_shutdown("Internal error: invalid mode.");
      return -1;
    }
    // ...
```